tools/fuchsia/build_fuchsia_artifacts.py

```diff
@@ -52,41 +74,26 @@ def RemoveDirectoryIfExists(path):
   else:
     shutil.rmtree(path)
 
-def CopyFiles(source, destination):
-  try:
-    shutil.copytree(source, destination)
-  except OSError as error:
-    if error.errno == errno.ENOTDIR:
-      shutil.copy(source, destination)
-    else:
-      raise
 
 def CopyToBucket(source, destination):
   source = os.path.join(_out_dir, source)
-  destination = os.path.join(_bucket_directory, destination)
+  temp_artifact_dir = tempfile.mkdtemp()
+
+  GatherArtifacts(source, temp_artifact_dir)
+  pm_bin = GetPMBinPath()
+  key_path = os.path.join(_script_dir, 'development.key')
 
-  assert os.path.exists(source), '"%s" does not exist!' % source
+  destination = os.path.join(_bucket_directory, destination)
+  CreateFarPackage(pm_bin, temp_artifact_dir, key_path, destination)
 
-  CopyFiles(source, destination)
 
 def BuildBucket():
   RemoveDirectoryIfExists(_bucket_directory)
 
-  CopyToBucket('fuchsia_debug/flutter_patched_sdk', 'flutter/debug/flutter_patched_sdk')
-  CopyToBucket('fuchsia_profile/flutter_patched_sdk', 'flutter/profile/flutter_patched_sdk')
-  CopyToBucket('fuchsia_release/flutter_patched_sdk', 'flutter/release/flutter_patched_sdk')
-
-  CopyToBucket('fuchsia_debug/flutter_runner', 'flutter/debug/flutter_runner')
-  CopyToBucket('fuchsia_profile/flutter_runner', 'flutter/profile/flutter_runner')
-  CopyToBucket('fuchsia_release/flutter_runner', 'flutter/release/flutter_runner')
-
-  CopyToBucket('fuchsia_debug/dart_runner', 'flutter/debug/dart_runner')
-  CopyToBucket('fuchsia_profile/dart_runner', 'flutter/profile/dart_runner')
-  CopyToBucket('fuchsia_release/dart_runner', 'flutter/release/dart_runner')
+  CopyToBucket('fuchsia_debug/', 'flutter/debug/')
```

Comment:
Won't this contain a ton of unnecessary stuff? Including object files and other generated intermediates?Basically the entire out/ directory?

Reply:
Nope. `CopyToBucket` delegates to `GatherArtifacts` which limits to the ones specified in https://github.com/flutter/engine/pull/10004/files#diff-7e0575351ad52a9482431aafd699b6e2R18.